src/be_db_block.erl

```diff
@@ -144,15 +144,21 @@ maybe_write_snapshot(Height, SnapshotHash, SnapshotDir, Chain) ->
         end,
     Filename = filename:join([SnapshotDir, io_lib:format("snap-~p", [Height])]),
     ok = blockchain:save_bin_snapshot(Filename, BinSnap),
+    ok = blockchain:save_compressed_bin_snapshot(Filename ++ ".gz", {file, Filename}),
     {ok, FileSHA} = blockchain:hash_bin_snapshot(BinSnap),
     Size = blockchain:size_bin_snapshot(BinSnap),
-    LatestBin = jsone:encode(#{
+    LatestBin0 = jsone:encode(#{
         height => Height,
         file_size => Size,
         file_hash => base64url:encode(FileSHA),
         hash => base64url:encode(SnapshotHash)
     }),
     Latest = filename:join([SnapshotDir, "latest-snap.json"]),
+    LatestBin = case blockchain:maybe_get_compressed_snapdata(Filename) of
+                    undefined -> LatestBin0;
+                    {ok, CSz, CHash} -> LatestBin0#{compressed_hash => CHash,
```

Comment:
and here you are adding to what you think is a map? I don't think that works

Reply:
Ah you are right about that - thanks